### ISSUE TITLE
Issue with SWA Implementation 

### ISSUE THREAD

**Issue by wjmaddox** [NajibYavari/DD2412#1]
Hi,

Thanks for choosing our paper for the NeurIPS reproducibility challenge. A couple of quick comments - we don't find it terribly surprising that averaging, SWA, and SWAG don't work particularly well when the loss blows up. After all, the theory we mention in the paper ought to only hold around minima - if the loss blows up you will really not be capturing the posterior in any sense. It's also disappointing that weight decay and SGD with momentum are implemented so differently in Tensorflow and Pytorch, but that's beyond our control.

I started looking into your implementation of SWA and noticed that you seem to be incorporating at least one model from initialization into the SWA solution: https://github.com/DD2412-Final-Projects/swag-reproduced/blob/2aef0a862abac8e1d0d01558c19ad1eeee9d27eb/train_swag.py#L167

I'd suggest multiplying everything by zero at the beginning to ensure that the initial models don't end up in the average.

Cheers, 
Wesley

P.S. A reasonable looking SWA implementation can be found here: https://github.com/JGuillaumin/swa-tf reproducing many of the results from the first paper.
